testjson/format.go

```diff
@@ -73,6 +75,55 @@ func standardJSONFormat(out io.Writer) EventFormatter {
 	})
 }
 
+func testDoxFormat(out io.Writer, opts FormatOptions) EventFormatter {
+	buf := bufio.NewWriter(out)
+	type Result struct {
+		Event    TestEvent
+		Sentence string
+	}
+	getIcon := icon
+	if opts.UseHiVisibilityIcons {
+		getIcon = iconHiVis
+	}
+	results := map[string][]Result{}
+	return eventFormatterFunc(func(event TestEvent, exec *Execution) error {
+		switch {
+		case event.PackageEvent():
+			if !event.Action.IsTerminal() {
+				return nil
+			}
+			if opts.HideEmptyPackages && len(results[event.Package]) == 0 {
```

Comment:
I added `opts.HideEmptyPackages` here to match the behaviour of other formats. Print the empty package by default, but `--format-hide-empty-pkg` can be set to hide these (for projects that have lots of empty packages).